src/HooglePlus/GHCChecker.hs

```diff
@@ -121,6 +124,37 @@ checkStrictness tyclassCount body sig modules =
         (\(SomeException _) -> return False)
         (checkStrictness' tyclassCount body sig modules)
 
+check :: Goal -> SearchParams -> Chan Message -> Chan Message -> IO ()
+check goal searchParams solverChan checkerChan = catch
+    (evalStateT (check_ goal searchParams solverChan checkerChan) emptyFilterState)
+    (\err ->
+        writeChan checkerChan (MesgLog 0 "error" (show err)) >>
+        writeChan checkerChan (MesgClose (CSError err)))
+
+check_ :: MonadIO m => Goal -> SearchParams -> Chan Message -> Chan Message -> FilterTest m ()
+check_ goal searchParams solverChan checkerChan = do
+    msg <- liftIO $ readChan solverChan
+    handleMessages solverChan checkerChan msg
+    return ()
+    where
+        handleMessages solverChan checkChan msg =
+            case msg of
+                (MesgP (program, _)) -> do
+                    programPassedChecks <- executeCheck program
+                    show <$> get >>= log
+                    if programPassedChecks then bypass >> next else next
+                (MesgClose _) -> bypass
+                _ -> bypass >> next
+
+            where
+                next = (liftIO . readChan) solverChan >>= handleMessages solverChan checkerChan
+                bypass = liftIO $ writeChan checkerChan msg
+                log msg = liftIO $ writeChan checkerChan (MesgLog 0 "checker" msg)
```

Comment:
Not all messages should be on `0`. Think of it like:
- `0`: `CRITICAL / ALWAYS`
- `1`: `WARN`
- `2`: `INFO`
- `3`: `DEBUG`
- `4+`: `VERY VERBOSE`